### PR TITLE
implement the save model method

### PR DESCRIPTION
**Pull request by steveoni**
The implementation enable the saving of `svm model`

```js
const model_save = await svm.save("svm.model");
```

The above command will create a text file name `svm.model` and store in the necessary svm parameters and weights.

Since in pushing the code, the adjustment made to the `libsvm` library is not available in the code. Here are the changes made to some of the files.

In `svm.cpp` the  `svm_save_model` was updated

```cpp
#include "emscripten.h"
int svm_save_model(const svm_model *model, const char *model_file_name)
{
	EM_ASM(
		FS.mkdir('/working');
		FS.mount(NODEFS, {root : '.'}, '/working'););

	const char *cwd = "/working/";

	char *fulltext = (char *)malloc(strlen(cwd) + strlen(model_file_name) + 1);

	strcpy(fulltext, cwd);
	strcat(fulltext, model_file_name);

	FILE *fp = fopen(fulltext, "w");
        // the remaining lines of code below are the same
```
first change to `svm.cpp` is the argument order it was change from `int svm_save_model(const char *model_file_name, const svm_model *model, )` to `int svm_save_model(const svm_model *model, const char *model_file_name)`. For some weird reason, the former affects the code.

In order to access the `Node` filesysystem, we need to Mount the Node filesystem using `EM_ASM(...)` to replace the Emscripten virtual file system.

Due to the changes made to `svm.cpp` , changes are made to some files

`svm.h`
```c
int svm_save_model(const struct svm_model *model, const char *model_file_name);
```

`libsvm-wasm.c`

```c
bool save_model(svm_model *model, const char *model_file_name)
{
    int success = svm_save_model(model, model_file_name);
    if (success < 0)
        return false;
    return true;
}
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -156,6 +156,23 @@ class SVM {
...
+        if (this.modelPointer == null) {
+            console.error("Model should be trained first");
+            return;
+        }
+
+        var buffer = libsvm._malloc(name.length+1);
+
+        // Write the string to memory
+        libsvm.stringToUTF8(name, buffer,name.length+1);
+        // const pred: boolean = libsvm._save_model(this.modelPointer, buffer);
```
Comment (WenheLI) [inline src/index.ts]: What's the purpose of this comment?

File: src/index.ts [hunk 2]
```diff
@@ -156,6 +156,23 @@ class SVM {
...
+
+        if (this.modelPointer == null) {
+            console.error("Model should be trained first");
+            return;
+        }
+
+        var buffer = libsvm._malloc(name.length+1);
+
+        // Write the string to memory
+        libsvm.stringToUTF8(name, buffer,name.length+1);
```
Comment (WenheLI) [inline src/index.ts:170]:
```suggestion
        libsvm.stringToUTF8(name, buffer, name.length+1);
```

File: libsvm-wasm.c [hunk 3]
```diff
@@ -17,22 +17,26 @@ typedef struct svm_parameter svm_parameter;
...
-//     return true;
-// }
+EMSCRIPTEN_KEEPALIVE
+bool save_model(svm_model *model, const char *model_file_name)
+{
+    int success = svm_save_model(model, model_file_name);
+    if (success < 0)
+        return false;
+    return true;
+}
```
Comment (WenheLI) [inline libsvm-wasm.c]: This seems to be different from what you have commented in the issue.

File: src/index.ts (hunk 1 above)
Comment (steveoni) [reply src/index.ts]: I scrape the call to `libscm._save_model` and i forgot to clear it out after working.

File: libsvm-wasm.c (hunk 3 above)
Comment (steveoni) [reply libsvm-wasm.c]: it's thesame
```c
bool save_model(svm_model *model, const char *model_file_name)
{
    int success = svm_save_model(model, model_file_name);
    if (success < 0)
        return false;
    return true;
}
```

Comment (steveoni) [reply libsvm-wasm.c]: can you make this more clear, I don't quite understand what you mean

Comment (WenheLI) [reply libsvm-wasm.c]: Humm I get it! You implemented `svm_save_model` for wasm in the `libsvm/libsvm.cpp`. I think we should implement it under `libsvm-wasm.c`.

What's your insight about it?

Comment (steveoni) [reply libsvm-wasm.c]: I've made some new commit. The full implementation is now moved into `lib-wasm.c` also I've been able to implement the `load_model` and `evaluation` method

**Comment by WenheLI** [comment]
@steveoni - Thanks!!! This is really amazing work! The only thing that came into my mind is that the saveModel and loadModel function can only work for the node environment for now. 

I will handle the Browser part after this is merged.

At last, would you mind cleaning the comments you left? So that I could merge it!

**Comment by steveoni** [comment]
The comment is now removed.

**Comment by WenheLI** [comment]
LGTM
Thanks

**Pull request merged**